nucypher/policy/policies.py

```diff
@@ -407,288 +241,339 @@ def hrac(self) -> bytes:
 
-        return arrangement_is_accepted
+        self._enact_arrangements(network_middleware=network_middleware,
+                                 arrangements=arrangements,
+                                 publish_treasure_map=publish_treasure_map)
 
-    def make_arrangements(self,
-                          network_middleware: RestMiddleware,
-                          handpicked_ursulas: Optional[Set[Ursula]] = None,
-                          discover_on_this_thread: bool = True,
-                          *args, **kwargs,
-                          ) -> None:
+        treasure_map = self._make_treasure_map(network_middleware=network_middleware,
+                                               arrangements=arrangements)
+        treasure_map_publisher = self._make_publisher(treasure_map=treasure_map,
+                                                      network_middleware=network_middleware)
+        revocation_kit = RevocationKit(treasure_map, self.alice.stamp)
 
-        sampled_ursulas = self.sample(handpicked_ursulas=handpicked_ursulas,
-                                      discover_on_this_thread=discover_on_this_thread)
+        enacted_policy = EnactedPolicy(self._id,
+                                       self.hrac,
+                                       self.label,
+                                       self.public_key,
+                                       treasure_map,
+                                       treasure_map_publisher,
+                                       revocation_kit,
+                                       self.alice.stamp)
 
-        if len(sampled_ursulas) < self.n:
-            raise self.MoreKFragsThanArrangements(
-                "To make a Policy in federated mode, you need to designate *all* '  \
-                 the Ursulas you need (in this case, {}); there's no other way to ' \
-                 know which nodes to use.  Either pass them here or when you make ' \
-                 the Policy.".format(self.n))
+        if publish_treasure_map is True:
+            enacted_policy.publish_treasure_map()
 
-        # TODO: One of these layers needs to add concurrency.
-        self._propose_arrangements(network_middleware=network_middleware,
-                                   candidate_ursulas=sampled_ursulas,
-                                   *args, **kwargs)
+        return enacted_policy
 
-        if len(self._accepted_arrangements) < self.n:
-            formatted_offenders = '\n'.join(f'{u.checksum_address}@{u.rest_url()}' for u in sampled_ursulas)
-            raise self.Rejected(f'Selected Ursulas rejected too many arrangements'
-                                f'- only {len(self._accepted_arrangements)} of {self.n} accepted.\n'
-                                f'Offending nodes: \n{formatted_offenders}\n')
+    @abstractmethod
+    def _blockchain_signer(self) -> Optional[Callable[[bytes], bytes]]:
```

Comment:
`FederatedPolicy` doesn't have any signers, so this function either returns an actual blockchain signer, or nothing.

Reply:
I understand the need and logic for the abstract signer method - but I was referring to the name of the method. Having `blockchain` in the method name, causes `FederatedPolicy._blockchain_signer()` to be defined, which is an oxymoron.

Something other than `blockchain` which is the direct antithesis of `Federated`, would be better, like `_signer()`, or `_transaction_signer()` (since you use `transaction` as a term in other areas of `Policy` eg. `_max_enactment_payload()` and `_enact_arrangements()`).

Reply:
After some thought, I removed `_blockchain_signer()` completely and instead add the signature in the overridden `BlockchainPolicy._make_treasure_map()`. This seems more logical.

Reply:
👍 - even better